From 71b2c877c99da73d96034782c5154c8087825171 Mon Sep 17 00:00:00 2001
From: Tao Ren <taoren@meta.com>
Date: Mon, 25 Sep 2023 17:45:28 +0000
Subject: [PATCH]  ARM: dts: aspeed: Add Facebook tahan BMC


tahan is Facebook's next generation switch platform with an
 AST2600 BMC integrated for health monitoring purpose.

---
 .../boot/dts/aspeed-bmc-facebook-tahan.dts    | 92 +++++++++++++++++++
 1 file changed, 92 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-facebook-tahan.dts

diff --git a/arch/arm/boot/dts/aspeed-bmc-facebook-tahan.dts b/arch/arm/boot/dts/aspeed-bmc-facebook-tahan.dts
new file mode 100644
index 000000000000..611659ff8a24
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-facebook-tahan.dts
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0+
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+/dts-v1/;
+
+#include "ast2600-facebook-netbmc-common.dtsi"
+
+/ {
+	model = "Facebook tahan BMC";
+	compatible = "facebook,tahan-bmc", "aspeed,ast2600";
+
+	chosen {
+		stdout-path = &uart1;
+	};
+
+	spi_gpio: spi-gpio {
+		num-chipselects = <1>;
+		cs-gpios = <&gpio0 ASPEED_GPIO(X, 0) GPIO_ACTIVE_LOW>;
+	};
+};
+
+&peci0 {
+	status = "okay";
+
+	peci-client@30 {
+		compatible = "intel,peci-client";
+		reg = <0x30>;
+	};
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&lpc_ctrl {
+	status = "okay";
+	flash = <&spi1>;
+};
+
+&kcs1 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xca0>;
+};
+
+&kcs2 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xca8>;
+};
+
+&kcs3 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xca2>;
+};
+
+/* JTAG1 will be used for COMe FPGA upgrade. */
+&jtag1 {
+	status = "okay";
+};
+
+/* SPI1 will be used for BIOS and IOB/FPGA flash upgrades. */
+&spi1 {
+	status = "okay";
+
+	reg = < 0x1e630000 0xc4
+			0x30000000 0x2000000 >;
+
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+		label = "spi1.0";
+		spi-rx-bus-width = <1>;
+		spi-max-frequency = <5000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash1@0 {
+				reg = <0x0 0x2000000>;
+				label = "spi-flash";
+			};
+		};
+	};
+};
-- 
2.25.1
